modules/core/src/test/scala/io/janstenpickle/hotswapref/HotswapRefSpec.scala

```diff
@@ -48,11 +48,11 @@ class HotswapRefSpec extends AnyFlatSpec with Matchers with TestInstances {
       )
       _ <- Resource.eval(IO.sleep(1.second)) // give it a chance to update
       _ <- Resource.eval(hotswap.swap(Resource.pure(ref1)).start)
-      _ <- Resource.eval(IO.sleep(1.second)) // give it a chance to update
+      _ <- Resource.eval(IO.sleep(1.second)) // give it a chance to allocate a new resource and refresh the holder ref
       _ <- Resource.eval(hotswap.access.use(_.update("test1" :: _)))
     } yield (ref0.get, ref1.get, releaseSignal.get)).use(_.tupled)
```

Comment:
Expecting this `release.get` being `false` at the end of the test somehow goes against my intuition of what `HotswapRef` tries to achieve and what is mentioned in the scaladoc. Particularly this line is confusing to me:

```scala
  * As with calls to [[swap]] on [[cats.effect.std.Hotswap]], [[swap]] will block until the previous
  * [[cats.effect.kernel.Resource]] is finalized.
```

Reply:
IMO, the scaladoc is correct and the test is not.

Reply:
This is how I would rewrite this test and how I understand it based on the `HotswapRef` scaladoc.

```scala
  it should "block on swap while handles to previous resource are held" in {
    val test = (for {
      ref0 <- ref
      ref1 <- ref
      releaseSignal <- Resource.eval(Ref.of[IO, Boolean](false))
      hotswap <- HotswapRef[IO, Ref[IO, List[String]]](Resource.pure(ref0))
      gate1 <- Resource.eval(Deferred[IO, Unit])
      _ <- Resource.eval(
        hotswap.access.use(_.update("test0" :: _) >> gate1.complete(()) >> IO.sleep(10.seconds) >> releaseSignal.set(true)).start
      )
      _ <- Resource.eval(gate1.get) // the update succeeded
      gate2 <- Resource.eval(Deferred[IO, Unit])
      _ <- Resource.eval((hotswap.swap(Resource.pure(ref1)) >> gate2.complete(())).start)
      _ <- Resource.eval(gate2.get) // the swap is complete
      _ <- Resource.eval(hotswap.access.use(_.update("test1" :: _)))
    } yield (ref0.get, ref1.get, releaseSignal.get)).use(_.tupled)

    val res = test.unsafeRunSync()

    // each ref has been used
    res._1.size should be(1)
    res._2.size should be(1)
    // access isn't blocked by hotswap
    res._1.head should be("test0")
    res._2.head should be("test1")
    // original resource isn't yet released
    res._3 should be(false) // this should be `true`
  }
```